libc/string_lenchr.c

```diff
@@ -96,61 +99,255 @@ TEST(string_len, ascii)
+	TEST_ASSERT_EQUAL_INT(0, strspn(emptySet, "abc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("", "abc"));
+	TEST_ASSERT_EQUAL_INT(0, strspn("", "abc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("", ""));
+	TEST_ASSERT_EQUAL_INT(0, strspn("", ""));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("\0abc", "abc"));
+	TEST_ASSERT_EQUAL_INT(0, strspn("\0abc", "abc"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("\0abc", ""));
+	TEST_ASSERT_EQUAL_INT(0, strspn("\0abc", ""));
+}
+
+
+TEST(string_spn, mixed_order)
+{
+
+	char *asciiSet = testdata_createCharStr(BUFF_SIZE),
+		 strHolder[BUFF_SIZE] = { 0 },
+		 supportSet[BUFF_SIZE] = { 0 };
+	int i;
+
+	asciiSet[BUFF_SIZE - 1] = 0;
+
+	/*
+	 * In those cases strcspn look for strings which not equal on the other hand strspn
+	 * looking for equal strings
+	 */
+	TEST_ASSERT_EQUAL_INT(0, strcspn("a", "a"));
+	TEST_ASSERT_EQUAL_INT(1, strspn("a", "a"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("abc", "a"));
+	TEST_ASSERT_EQUAL_INT(1, strspn("abc", "a"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("abc", "cba"));
+	TEST_ASSERT_EQUAL_INT(3, strspn("abc", "cba"));
+
+	TEST_ASSERT_EQUAL_INT(0, strcspn("abc", "abc1"));
+	TEST_ASSERT_EQUAL_INT(3, strspn("abc", "abc1"));
+
+	memset(strHolder, 'a', BUFF_SIZE - 3);
+	strHolder[BUFF_SIZE - 2] = 'b';
+	memset(supportSet, 'b', BUFF_SIZE - 3);
+	supportSet[BUFF_SIZE - 2] = 'a';
+
+	TEST_ASSERT_EQUAL_INT(BUFF_SIZE - 3, strcspn(strHolder, supportSet));
+	TEST_ASSERT_EQUAL_INT(0, strspn(strHolder, supportSet));
+
+	TEST_ASSERT_EQUAL_INT(BUFF_SIZE - 3, strcspn(supportSet, strHolder));
+	TEST_ASSERT_EQUAL_INT(0, strspn(supportSet, strHolder));
+
+	memcpy(supportSet, asciiSet, BUFF_SIZE - 1);
+
+	for (i = 0; i < BUFF_SIZE - 1; i++) {
+
+		/*
+		 * Checking if strcspn and strspn will search
+		 * the whole string before returning value and
+		 * he found an element of impact on the output
+		 */
+
+		/* Setting only the first element of set to values from asciiSet */
+		supportSet[0] = asciiSet[i];
+		TEST_ASSERT_EQUAL_INT(0, strcspn(asciiSet, supportSet));
+
+		/* After passing halfway through strspn will start to grow up again */
+		if (i < (BUFF_SIZE - 1) / 2) {
+			TEST_ASSERT_EQUAL_INT(BUFF_SIZE - 1 - i, strspn(asciiSet, supportSet));
+		}
+		else {
+			TEST_ASSERT_EQUAL_INT(i + 1, strspn(asciiSet, supportSet));
+		}
+
+		/* Changing approach to setup elements, setting them up dynamically depending on the value from end to start */
+		supportSet[0] = asciiSet[0];
+		supportSet[BUFF_SIZE - 2 - i] = asciiSet[i];
+
+		printf("%c\n", asciiSet[i]);
+
+		TEST_ASSERT_EQUAL_INT(0, strcspn(asciiSet, supportSet));
+
+		/* Same case as before */
+		if (i < (BUFF_SIZE - 1) / 2) {
+			TEST_ASSERT_EQUAL_INT(BUFF_SIZE - 2 - i, strspn(asciiSet, supportSet));
+		}
+		else {
+			TEST_ASSERT_EQUAL_INT(i + 1, strspn(asciiSet, supportSet));
+		}
+	}
+
+	free((void *)asciiSet);
+}
+
+
+TEST(string_spn, out_of_set)
+{
+	char pangram[45] = " The quick brown fox jumps over the lazy dog",
```

Comment:
Why 45? Yoo could declare it this way:
```suggestion
	const char pangram[] = " The quick brown fox jumps over the lazy dog",
```

Please remember about `const`.